ocs_ci/utility/workloadfixture.py

```diff
@@ -0,0 +1,168 @@
+            alert_list (list): List to be populated with alerts
+        """
+        prometheus = PrometheusAPI()
+        logger.info('Logging of all prometheus alerts started')
+        while info.get('run'):
+            alerts_response = prometheus.get(
+                'alerts',
+                payload={
+                    'silenced': False,
+                    'inhibited': False
+                }
+            )
+            msg = f"Request {alerts_response.request.url} failed"
+            assert alerts_response.ok, msg
+            for alert in alerts_response.json().get('data').get('alerts'):
+                if alert not in alert_list:
+                    logger.info(f"Adding {alert} to alert list")
+                    alert_list.append(alert)
+            time.sleep(3)
+        logger.info('Logging of all prometheus alerts stopped')
+
+    # check if file with results for this operation already exists
+    # if it exists then use it
+    if os.path.isfile(result_file) and os.access(result_file, os.R_OK):
+        logger.info(
+            f"File {result_file} already created."
+            f" Trying to use it for tests..."
+        )
+        with open(result_file) as open_file:
+            results = json.load(open_file)
+            # indicate that we are not going to execute the workload, but
+            # just reuse measurement from earlier run
+            results['first_run'] = False
+        logger.info(
+            f"File {result_file} loaded. Content of file:\n{results}"
+        )
+
+    # if there is no file with results from previous run
+    # then perform operation measurement
+    else:
+        logger.info(
+            f"File {result_file} not created yet. Starting measurement..."
+        )
+        if not measure_after:
+            start_time = time.time()
+
+        # init logging thread that checks for Prometheus alerts
+        # while workload is running
+        # based on https://docs.python.org/3/howto/logging-cookbook.html#logging-from-multiple-threads
+        info = {'run': True}
+        alert_list = []
+
+        logging_thread = threading.Thread(
+            target=prometheus_log,
+            args=(info, alert_list)
+        )
+        logging_thread.start()
+
+        try:
+            result = operation()
+        except Exception as ex:
+            # When the operation (which is being measured) fails, we need to
+            # make sure that alert harvesting thread ends and (at least)
+            # alerting data are saved into measurement dump file.
+            result = None
+            logger.error("exception raised during measured operation: %s", ex)
+            # Additional waiting for the measurement purposes is no longer
+            # necessary, and would only confuse anyone observing the failure.
+            minimal_time = 0
+            # And make sure the exception is properly processed by pytest (it
+            # would make the fixture fail).
+            raise(ex)
```

Comment:
```suggestion
            raise
```

Reply:
why?

Reply:
Explained https://github.com/red-hat-storage/ocs-ci/pull/1714/files#r406111243 - IIRC the raise will properly copy origin traceback.

Reply:
And as you do not modify the exception anyhow, the simple re-raising exception is defined in pythons doc like this as well: https://docs.python.org/3/tutorial/errors.html#raising-exceptions. 
`If you need to determine whether an exception was raised but don’t intend to handle it, a simpler form of the raise statement allows you to re-raise the exception`

Reply:
This code:
- is actually re raising the exception in explicit way on purpose, because in few lines above, I'm using a reference to the exception for the logging purposes
- no traceback information is actually lost by this explicit re raise